### PR TITLE
fix: Allow dynamic input shapes when exporting nnx.Conv with padding="REPEAT"

### PR DESCRIPTION
**Pull request by jlindermeir**
This PR adds a regression test for #210 and attempts a fix. To ensure transparency: **All code was written by Codex** (with gpt-5.4 xhigh). I ensured that the agent is reasonably familiar with the repo and added the test before writing the fix, but I would not consider myself familiar enough with the codebase to guarantee that the approach is fully correct. Additionally, I confirmed that the fixed version was able to successfully export my model with the affected `nnx.Conv` modules. Since this is only touching a single file of the implementation, maybe it's worth reviewing?

Codex-generated summary below:

---

## Summary

- Fix `nnx.Conv(..., padding="REFLECT")` export when the convolution dimension is symbolic.
- Add a regression test for issue `#210`.

## Problem

Issue `#210` reports that exporting an `nnx.Conv` with `padding="REFLECT"` fails when the input length is symbolic, even though the same model works with fixed input sizes.

The failing path goes through the reflect-padding helper emitted by JAX, which contains `lax.slice` operations with symbolic bounds such as `N`, `N + 1`, and `N - 1`. The current `lax.slice` lowering tried to coerce those bounds into constant `int64` initializers via `np.asarray(..., dtype=np.int64)`, which raises `InconclusiveDimensionOperation` for symbolic dimensions.

## Fix

- Keep the existing constant fast path for ordinary static slice bounds.
- Normalize slice indices so constant-looking symbolic values still fold to plain Python ints when possible.
- Lower non-constant symbolic `start_indices` and `limit_indices` through `ctx.dim_expr_lowerer` instead of forcing them into initializers.
- Leave the rest of the `Slice` lowering behavior unchanged, including axis handling and output metadata stamping.

## Regression Coverage

- Add `tests/extra_tests/test_issue_210_dynamic_reflect_padding.py`.
- The new test reproduces the minimal issue pattern and asserts that `to_onnx(...)` succeeds for `nnx.Conv(..., padding="REFLECT")` with a symbolic length constraint `N >= 2`.

## Validation

- `poetry run pytest -q tests/extra_tests/test_issue_210_dynamic_reflect_padding.py`
- `poetry run pytest -q tests/primitives/test_nnx.py -k conv_2d_reflect_padding`
- `poetry run pytest -q tests/primitives/test_lax.py -k slice_test1`
- `poetry run python scripts/generate_tests.py`
- `poetry run pytest -n auto --dist=loadscope tests/primitives tests/extra_tests tests/test_gpt_oss_components.py --ignore=tests/extra_tests/examples`

## Baseline Status

The reduced local suite still has the same three pre-existing failures as before this change:

- `tests/primitives/test_jnp.py::Test_nanmedian::test_jnp_nanmedian_basic`
- `tests/primitives/test_jnp.py::Test_nanquantile::test_jnp_nanquantile_basic`
- `tests/primitives/test_jnp.py::Test_nanpercentile::test_jnp_nanpercentile_basic`

The post-change reduced suite result was:

- `2894 passed`
- `3 failed`

No new failures were introduced in that local baseline run. The heavier integration/example suites were not rerun locally.

### CODE REVIEW

**Comment by jlindermeir** [comment]
Closing due to already merged fixed for the same issue by @enpasos
